### DISCUSSION TITLE
Getting a vector of development ages

### DISCUSSION THREAD

**Discussion by SleepyBob** [casact/chainladder-python#544] [Q&A] [answered]
my_tri.latest_diagonal will return the diagonal with the origin year as the index:
 _____          2024
2020  20000
2021  15000
2022  10000
2023  5000
2024  1000

How can I also get the corresponding development ages? e.g.
_____           Development   2024
2020    56                   20000
2021    44                   15000
2022    32                   10000
2023    20                     5000
2024    8                       1000

**Comment by kennethshsu** [comment, marked as answer]
Hi @SleepyBob,

I don't know that this feature is supported currently. @jbogaardt?

However, the ages can be quickly calculated as a workaround.

```python
import pandas as pd
import chainladder as cl

xyz = cl.load_sample("xyz")
(xyz["Incurred"].latest_diagonal.valuation.year - xyz["Incurred"].origin.year) * 12 + 12
```

```python
Index([132, 120, 108, 96, 84, 72, 60, 48, 36, 24, 12], dtype='int64')
```

Does this help?

**Reply by SleepyBob** [reply]
Thanks, I think I can use that as a workaround (though I think I have to pull the month from the valuation date instead of assuming 12).  This does seem like a fundamental piece of data, though.

**Reply by kennethshsu** [reply]
Yes and yes!

**Comment by henrydingliu** [comment]
Slightly longer alternative

```python
import pandas as pd
import chainladder as cl

xyz = cl.load_sample("xyz")
Exhibit = pd.DataFrame()
Exhibit['Incurred'] = xyz['Incurred'].latest_diagonal.to_frame(origin_as_datetime=False).iloc[:,0].fillna(0)
#change the indexing here if a tail estimator is added
age = xyz.development.iloc[::-1]
age.index = Exhibit.index
Exhibit['Age'] = age
Exhibit = Exhibit[['Age','Incurred']]
Exhibit
```
